test/fakeOrigin/httpService/rangerequest.go

```diff
@@ -0,0 +1,130 @@
+package httpService
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type httpRange struct {
+	startOffset int64
+	length      int64
+}
+
+const multipartSeparator = "3d6b6a416f9b5fakeOrigin"
+
+func parseRange(rawRange string, size int64) ([]httpRange, error) {
+	var out []httpRange
+	if rawRange == "" {
+		return nil, nil
+	}
+	if !strings.HasPrefix(rawRange, "bytes=") {
+		return nil, errors.New("invalid range")
+	}
+	for _, r := range strings.Split(strings.TrimPrefix(rawRange, "bytes="), ",") {
+		r = strings.TrimSpace(r)
+		if r == "" {
+			continue
+		}
+		if !strings.Contains(r, "-") {
+			return nil, errors.New("invalid range")
+		}
+		offsets := strings.Split(r, "-")
+		if len(offsets) > 2 || len(offsets) <= 0 {
+			return nil, errors.New("invalid range")
+		}
+		start := strings.TrimSpace(offsets[0])
+		startI, err := strconv.ParseInt(start, 10, 64)
+		if err != nil || startI >= size || startI < 0 {
```

Comment:
You can collapse the negative number check into the error if you do `strconv.ParseUint` instead. It also gives back a `uint64` which is probably a more appropriate analogue of `offset_t`.

Reply:
Addressed with new commits